gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/notice/RouteWithDuplicateStopNotice.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.notice;
+
+import com.google.common.collect.ImmutableMap;
+
+public class RouteWithDuplicateStopNotice extends Notice {
```

Comment:
I think `TripWithDuplicateStopNotice` may be a better name, as that is what is actually causing the error.

The reason we use route ID to deduplicate the notices is that if we have 1 route with 100 identical trips that all have the same error, we don't want to spam the user with warnings.

However, the error isn't actually in the route. It could be caused by a few things, including:
- 2 different stops in stops.txt having the same name
- The order of stops being incorrect for a single trip in stop_times.txt
- etc.

The trip + the offending stop times in the trip are probably the most important things for the user to be able to debug the issue.

Reply:
But if this is the case then shouldn't we probably be generating a notice for each trip that so that they can debug the issues for all trips?

Reply:
Yeah, the deduplication based on route ID is one of those 'more art than science' additions to the rule.

The idea is that if this notice appears for the same stops in many trips on the same route, then the stops are probably correct, but should be named differently. E.g. the feed has named two different stops in a city like `--->CBD--->CBD--->`, which would confuse users. If this is the case, then the user only needs to fix 'one' issue - the name of the stops in stops.txt.

These additions to rules are generally based on real data we see in GTFS feeds, + feedback from users of our validator. In particular, this addition was introduced because we were generating a _lot_ of erroneous warnings for this rule, and deduplicating based on route ID gave a better experience to users (although I agree it's definitely not perfect).

Reply:
Ok yes I see, this makes sense. I have changed the names of the files to reflect this now.